### ISSUE TITLE
Multiple Users Cause lambda.null_resource.archive[0] timestamp Differences

### ISSUE THREAD

**Issue by razorsedge** [terraform-aws-modules/terraform-aws-lambda#63]
When two or more users/workstations are applying this resource, the archive timestamp will be different, causing constant changes in tf state even without changes in code or running environment.

```
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

module.notify_teams.module.lambda.data.external.archive_prepare[0]: Refreshing state...
module.notify_teams.module.lambda.local_file.archive_plan[0]: Refreshing state... [id=d277e5cfd6f66a2c0a9c704e32216933827ed6d9]
module.notify_teams.module.lambda.null_resource.archive[0]: Refreshing state... [id=5491149952268672725]
module.notify_teams.data.aws_region.current: Refreshing state...
module.notify_teams.module.lambda.data.aws_caller_identity.current[0]: Refreshing state...
module.notify_teams.module.lambda.data.aws_iam_policy_document.assume_role[0]: Refreshing state...
module.notify_teams.data.aws_caller_identity.current: Refreshing state...
module.notify_teams.aws_cloudwatch_log_group.lambda[0]: Refreshing state... [id=/aws/lambda/notify-teams-dev-default]
module.notify_teams.module.lambda.data.aws_partition.current[0]: Refreshing state...
module.notify_teams.module.lambda.data.aws_region.current[0]: Refreshing state...
module.notify_teams.module.lambda.aws_iam_role.lambda[0]: Refreshing state... [id=lambda-notify-teams-dev-default]
module.notify_teams.data.aws_iam_policy_document.lambda[0]: Refreshing state...
module.notify_teams.module.lambda.aws_iam_policy.additional_json[0]: Refreshing state... [id=arn:aws:iam::841871640087:policy/notify-teams-dev-default]
module.notify_teams.module.lambda.aws_lambda_function.this[0]: Refreshing state... [id=notify-teams-dev-default]
module.notify_teams.module.lambda.aws_iam_policy_attachment.additional_json[0]: Refreshing state... [id=notify-teams-dev-default]
module.notify_teams.module.lambda.aws_lambda_permission.unqualified_alias_triggers["AllowExecutionFromSNS"]: Refreshing state... [id=AllowExecutionFromSNS]
module.notify_teams.module.lambda.aws_lambda_permission.current_version_triggers["AllowExecutionFromSNS"]: Refreshing state... [id=AllowExecutionFromSNS]
module.notify_teams.aws_sns_topic_subscription.sns_notify_teams[0]: Refreshing state... [id=arn:aws:sns:us-west-2:841871640087:lms-dev-default:62bbf250-a141-4290-88c6-f3a1294f1f77]

------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create
  ~ update in-place
-/+ destroy and then create replacement

Terraform will perform the following actions:

  # module.notify_teams.module.lambda.null_resource.archive[0] must be replaced
-/+ resource "null_resource" "archive" {
      ~ id       = "5491149952268672725" -> (known after apply)
      ~ triggers = { # forces replacement
            "filename"  = "builds/c823beae37d6fe8d0a9292c46f734a4a4bbed5001e033d2bf28b843a1b57fdfb.zip"
          ~ "timestamp" = "1601663561805170000" -> "1598884633000000000"
        }
    }

Plan: 1 to add, 0 to change, 1 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

module.notify_teams.module.lambda.null_resource.archive[0]: Destroying... [id=5491149952268672725]
module.notify_teams.module.lambda.null_resource.archive[0]: Destruction complete after 0s
module.notify_teams.module.lambda.null_resource.archive[0]: Creating...
module.notify_teams.module.lambda.null_resource.archive[0]: Provisioning with 'local-exec'...
module.notify_teams.module.lambda.null_resource.archive[0] (local-exec): Executing: ["python3" ".terraform/modules/notify_teams.lambda/package.py" "build" "--timestamp" "1598884633000000000" "builds/c823beae37d6fe8d0a9292c46f734a4a4bbed5001e033d2bf28b843a1b57fdfb.plan.json"]
module.notify_teams.module.lambda.null_resource.archive[0] (local-exec): Reused: builds/c823beae37d6fe8d0a9292c46f734a4a4bbed5001e033d2bf28b843a1b57fdfb.zip
module.notify_teams.module.lambda.null_resource.archive[0]: Creation complete after 0s [id=7770971292503942157]

Apply complete! Resources: 1 added, 0 changed, 1 destroyed.
Releasing state lock. This may take a few moments...
```

**Comment by antonbabenko** [comment]
This sounds like a known limitation on this module, the workaround is described in https://github.com/terraform-aws-modules/terraform-aws-lambda/issues/51#issuecomment-681978205.

**Comment by 123BLiN** [comment]
Hey @antonbabenko it seems that we are still having this issue, and the same as descibed - we have multiple users.
Setting environment variable does not help, but I have next logs:
```
Terraform will perform the following actions:

  # module.cloudwatch_monitoring.module.notify_slack.module.lambda.null_resource.archive[0] must be replaced
-/+ resource "null_resource" "archive" {
      ~ id       = "4213460612593619993" -> (known after apply)
      ~ triggers = { # forces replacement
            "filename"  = "builds/5263a8b288d3aa735f5e3f690df38b5c344b6148f53581dd46fd72f43a77a3d3.zip"
          ~ "timestamp" = "1599219425723994000" -> "<WARNING: Missing lambda zip artifacts wouldn't be restored>"
        }
    }

Plan: 1 to add, 0 to change, 1 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

module.cloudwatch_monitoring.module.notify_slack.module.lambda.null_resource.archive[0]: Destroying... [id=4213460612593619993]
module.cloudwatch_monitoring.module.notify_slack.module.lambda.null_resource.archive[0]: Destruction complete after 0s
module.cloudwatch_monitoring.module.notify_slack.module.lambda.null_resource.archive[0]: Creating...
module.cloudwatch_monitoring.module.notify_slack.module.lambda.null_resource.archive[0]: Provisioning with 'local-exec'...
module.cloudwatch_monitoring.module.notify_slack.module.lambda.null_resource.archive[0] (local-exec): Executing: ["python3" ".terraform/modules/cloudwatch_monitoring.notify_slack.lambda/terraform-aws-lambda-1.6.0/package.py" "build" "--timestamp" "<WARNING: Missing lambda zip artifacts wouldn't be restored>" "builds/5263a8b288d3aa735f5e3f690df38b5c344b6148f53581dd46fd72f43a77a3d3.plan.json"]
module.cloudwatch_monitoring.module.notify_slack.module.lambda.null_resource.archive[0] (local-exec): Reused: builds/5263a8b288d3aa735f5e3f690df38b5c344b6148f53581dd46fd72f43a77a3d3.zip
module.cloudwatch_monitoring.module.notify_slack.module.lambda.null_resource.archive[0]: Creation complete after 0s [id=426740811431562546]

Apply complete! Resources: 1 added, 0 changed, 1 destroyed.
Releasing state lock. This may take a few moments...
```

**Comment by antonbabenko** [comment]
Please update to use [v1.26.0](https://github.com/terraform-aws-modules/terraform-aws-lambda/releases/v1.26.0) where the hash is now computing differently from the way it was before (which will lead to recreation) but equally for multiple executors.

**Comment by antonbabenko** [comment]
This issue has been resolved in version 8.5.0 :tada:
